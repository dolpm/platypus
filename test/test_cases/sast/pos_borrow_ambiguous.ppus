pipe test |'a, 'b| [x: &'a int, y: &'b int] |> &'b int {
  if (@x == 2) {
    |> x;
  }
  |> y;
}

pipe main [] |> unit {
  int x = 4;
  while (x < 5) {
    int y = 4;
    mut &int w = &y;
    loop 0 -> 5 as z {
      w = (test <| [&x, &y]);
    }
  }
  |> ();
}
