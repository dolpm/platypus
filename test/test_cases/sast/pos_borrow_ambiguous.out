pipe test |> ['a, 'b] |> [x: &'a int, y: &'b int] |> &'b int {
{(id: test_with_args)
{(id: test_wrapper)
  if ((bool : (int : @(&'a int : x)) == (int : 2)))
  {(id: 5)
    |> (&'a int : x);
  }
  |> (&'b int : y);
}
}
}

pipe main |> [] |> [] |> unit {
{(id: main_with_args)
{(id: main_wrapper)
  int x <| (int : 4);
  while (bool : (int : x) < (int : 5))
  {(id: 1)
    int y <| (int : 4);
    mut &'_ int w <| (&'_ int : &(int : y));
    loop (int : 0) -> (int : 5) as (z,(int : 1))    {(id: 2)
      w <| (&'_ int : test <| [(&'_ int : &(int : x)), (&'_ int : &(int : y))]);
    }
  }
  |> (unit : ());
}
}
}
